kernel/include/kernel_arch_interface.h

```diff
@@ -130,20 +130,6 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
  */
 static inline void arch_switch(void *switch_to, void **switched_from);
 #else
-/**
- * Cooperatively context switch
```

Comment:
It would be a pity to lose this doc.

Reply:
It turns out that the doc was not getting generated in the first place. This latest version amends that.

Reply:
This header is filled with that kind of thing: prototypes that turn out to be #if'd out for all or almost all architectures because of inlining concerns.  Might not be a bad idea to do something special like a "ARCH_INTERFACE_INLINE" attribute that would fold that __DOXYGEN__ check in such that the doc generation was guaranteed to see the prototype but that arch could could override...